meeteval/wer/matching/cy_levenshtein.pyx

```diff
@@ -113,132 +121,114 @@ def levenshtein_distance(
-            )
-        last_start = t.first
-
-    last_start = 0
-    for t in hypothesis_timing:
-        if t.second < t.first:
-            raise ValueError(
-                f'The end time of an interval {t.second} must not be smaller than its begin time {t.first}, '
-                f'but the hypothesis violates this'
-            )
-        if t.first < last_start:
-            raise ValueError(
-                f'The start times of the annotations must be increasing, which they are not for the hypothesis. '
-                f'(found at least one interval where {t.first} < {last_start}) '
-                f'This might be caused by overlapping segments, see the (potential) previous warning.'
-            )
-        last_start = t.first
+cdef bool _timing_is_integer(timing):
+    return all(isinstance(t[0], int) and isinstance(t[1], int) for t in timing)
 
-    return reference, hypothesis, reference_timing, hypothesis_timing
+ctypedef fused timing_type:
+    int
+    double
 
-def time_constrained_levenshtein_distance(
+cdef bool _timings_sorted(vector[pair[timing_type, timing_type]] timing):
+    cdef timing_type last_begin = 0
+    cdef pair[timing_type, timing_type] t
+    for t in timing:
+        if t.first < last_begin:
+            return False
+        last_begin = t.first
+    return True
+
+
+cdef uint _time_constrained_levenshtein_distance(
         reference,  # list[int]
         hypothesis,  # list[int]
-        reference_timing,  # list[tuple[int, int]]
-        hypothesis_timing,  # list[tuple[int, int]]
-        cost_del: uint = 1,
-        cost_ins: uint = 1,
-        cost_sub: uint = 1,
-        cost_cor: uint = 0,
+        reference_timing: vector[pair[timing_type, timing_type]],  # list[tuple[int, int]]
+        hypothesis_timing: vector[pair[timing_type, timing_type]],  # list[tuple[int, int]]
+        cost_del = 1,
+        cost_ins = 1,
+        cost_sub = 1,
+        cost_cor = 0,
+        prune = 'auto'
 ):
     _validate_costs(cost_del, cost_ins, cost_sub, cost_cor)
-    # reference_timing, hypothesis_timing = times_to_int(reference_timing, hypothesis_timing)
-    reference, hypothesis, reference_timing, hypothesis_timing = _validate_inputs(
-        reference, hypothesis, reference_timing, hypothesis_timing
-    )
-
 
+    # Shortcut when at least one of reference or hypothesis is empty
     if len(reference) == 0:
         return len(hypothesis) * cost_ins
     if len(hypothesis) == 0:
         return len(reference) * cost_del
+
+    # Convert to vectors of integers for the C++ implementation
     reference, hypothesis = obj2vec(reference, hypothesis)
 
-    args = (reference, hypothesis,
+    # Dispatch to the correct implementation
+    if prune == 'auto':
+        prune = _timings_sorted(reference_timing) and _timings_sorted(hypothesis_timing)
+        logging.debug(
+            f'prune is set to "auto". '
```

Comment:
`f'Changed prune from "auto" to {prune!r}. '`

Reply:
I feel that your message could be understood wrongly. I shortened the message